website_and_docs/content/documentation/grid/advanced_features/external_datastore_sessions.en.md

```diff
@@ -0,0 +1,272 @@
+    * [postgresql](https://mvnrepository.com/artifact/org.postgresql/postgresql) which is needed to help us talk PostGreSQL database.
+* `sessions.toml` is the configuration file that we created earlier.
+
+
+## Redis backed Session Map
+
+We will spin off a Redis Cache docker container using a docker compose file.
+
+### Steps
+
+You can skip this step if you already have a Redis Cache instance available at your disposal.
+
+* Create a file named `docker-compose.yml` with its contents as below:
+
+```yaml
+version: '3.8'
+services:
+  redis:
+    image: redis:bullseye
+    restart: always
+    ports:
+      - "6379:6379"
+```
+
+We can now start our redis container by running:
+
+```bash
+docker-compose up -d
+```
+
+* Create a Selenium configuration file named `sessions.toml` with the below contents:
+
+```toml
+[sessions]
+scheme = "redis"
+implementation = "org.openqa.selenium.grid.sessionmap.redis.RedisBackedSessionMap"
+hostname = "localhost"
+port = 6379
+```
+
+_Note:_ If you plan to use an existing Redis Cache instance, then replace `localhost` and `6379` with the actual host and port number of your instance.
+
+* Below is a simple shell script (let's call it `distributed.sh`) that we will use to bring up our distributed grid.
+
+```bash
+SE_VERSION=4.6.0
+JAR_NAME=selenium-server-${SE_VERSION}.jar
+PUBLISH="--publish-events tcp://localhost:4442"
+SUBSCRIBE="--subscribe-events tcp://localhost:4443"
+SESSIONS="--sessions http://localhost:5556"
+SESSIONS_QUEUE="--sessionqueue http://localhost:5559"
+# Start event bus
+echo 'Starting Event Bus'
+java -jar $JAR_NAME event-bus $PUBLISH $SUBSCRIBE --port 5557 &
+#Start New Session Queue
+echo 'Starting New Session Queue'
+java -jar $JAR_NAME sessionqueue --port 5559 &
+#Start session map
+echo 'Starting sessions manager'
+java -jar $JAR_NAME \
+--ext $(coursier fetch -p org.seleniumhq.selenium:selenium-session-map-redis:${SE_VERSION}) \
+sessions $PUBLISH $SUBSCRIBE --port 5556 --config sessions.toml &
+#Start distributor
+echo 'Starting distributor'
+java -jar $JAR_NAME  distributor $PUBLISH $SUBSCRIBE $SESSIONS $SESSIONS_QUEUE --port 5553 --bind-bus false &
+#Start router
+echo 'Starting router'
+java -jar $JAR_NAME router $SESSIONS --distributor http://localhost:5553 $SESSIONS_QUEUE --port 4444 &
+#Start node
+echo 'Starting node'
+java -jar $JAR_NAME node $PUBLISH $SUBSCRIBE &
+```
+
+* At this point the current directory should contain the following files:
+    * `docker-compose.yml`
+    * `sessions.toml`
+    * `distributed.sh`
+
+* You can now spawn the grid by running `distributed.sh` shell script and quickly run a test. You will notice that the Grid now stores (inserts and then deletes) session information into the Redis instance. You can perhaps make use of a Redis GUI such as [TablePlus](https://tableplus.com/) to see them (Make sure that you have setup a debug point in your test, because the values will get deleted as soon as the test runs to completion)
```

Comment:
```suggestion
* You can now spawn the Grid by running `distributed.sh` shell script and quickly run a test. You will notice that the Grid now stores session information into the Redis instance. You can use a Redis GUI such as [TablePlus](https://tableplus.com/) to see them (make sure that you have setup a debug point in your test, because the values will get deleted as soon as the test runs to completion).
```